crates/matrix-sdk-crypto/src/identities/device.rs

```diff
@@ -141,6 +141,35 @@ impl Device {
         }
     }
 
+    /// Is this our own device?
+    pub fn is_our_own_device(&self) -> bool {
```

Comment:
> Is this method in the right place? It feels it would be more natural if it lived elsewhere and took a Device as a parameter.

I would say so. Making this a free standing method reduces discoverability.

> The aspect that makes me ask this is the fact that we have to fetch into the VerificationMachine in order to get a VerificationStore to finally arrive at our own Account, even though those things have little to do with the device represented by a particular Device instance.

While the layering might be improved later on, unless you want to get rid of the `Device::request_verification()` method, a `Device` has to keep a reference to the `VerificationMachine`. This state machine, to have the ability to start/request a verification has to have access to:

1. The device owners user identity, this is kept in the store
2. Our own device AKA the `Account`
3. Our own user identity

Given the above, I think it's quite normal that a `Device` has a method to check if this is indeed our own device. This also helps with rendering of a device list in clients, where your own `Device` is usually highlighted.